python/python/lance/arrow.py

```diff
@@ -11,14 +11,428 @@
+    """
+
+    import numpy as np
+
+    def to_tensor(
+        self, decoder: Optional[Callable[[pa.BinaryArray], np.ndarray]] = None
+    ):
+        """
+        Decode encoded images and return a FixedShapeImageTensorArray
+
+        Parameters
+        ----------
+        decoder : Callable[pa.binary()], optional
+            A function that takes a pa.binary() and returns a numpy.ndarray
+            or pa.fixed_shape_tensor. If not provided, will attempt to use
+            tensorflow and then pillow decoder in that order.
+
+        Returns
+        -------
+        FixedShapeImageTensorArray
+            Array of images as tensors
+
+        Examples
+        --------
+        >>> import os
+        >>> uris = [os.path.join(os.path.dirname(__file__), "../tests/images/1.png")]
+        >>> encoded_image_array = ImageURIArray.from_uris(uris).read_uris()
+        >>> encoded_image_array.to_tensor()
+        <lance.arrow.FixedShapeImageTensorArray object at 0x...>
+        [[42, 42, 42, 255]]
+        """
+        import numpy as np
+
+        if not decoder:
+
+            def pillow_decoder(images):
+                import io
+
+                from PIL import Image
+
+                return np.stack(
+                    Image.open(io.BytesIO(img)) for img in images.to_pylist()
+                )
+
+            def tensorflow_decoder(images):
+                import tensorflow as tf
+
+                decoded_to_tensor = tuple(
+                    tf.io.decode_image(img) for img in images.to_pylist()
+                )
+                return tf.stack(decoded_to_tensor, axis=0).numpy()
+
+            decoders = [
+                ("tensorflow", tensorflow_decoder),
+                ("PIL", pillow_decoder),
+            ]
+            for libname, decoder_function in decoders:
+                try:
+                    __import__(libname)
+                    decoder = decoder_function
+                    break
+                except ImportError:
+                    pass
+            else:
+                raise ValueError(
+                    "No image decoder available. Please either install one of "
+                    "tensorflow, pillow, or pass a decoder argument."
+                )
+
+        image_array = decoder(self.storage)
+        if isinstance(image_array, pa.FixedShapeTensorType):
+            shape = image_array.shape
+            arrow_type = image_array.storage_type
+            tensor_array = image_array
+        else:
+            shape = image_array.shape[1:]
+            arrow_type = pa.from_numpy_dtype(image_array.dtype)
+            tensor_array = pa.FixedShapeTensorArray.from_numpy_ndarray(image_array)
+
+        return pa.ExtensionArray.from_storage(
+            FixedShapeImageTensorType(arrow_type, shape), tensor_array.storage
+        )
+
+
+# TODO: add VariableShapeImageTensorType once pa.VariableShapeTensorArray is available
+class FixedShapeImageTensorArray(ImageArray):
```

Comment:
Have you tested this with a large image? I find the repr is really dangerous right now and can crash VS Code. Maybe we can implement a repr that more effectively truncates output?

Reply:
I like the idea of having the shape and format in the repr 👍

Reply:
Added https://github.com/lancedb/lance/pull/1272/commits/10c8b345cea13bf8427f3eb865952ea0d32ccf95